Patch-Source: https://github.com/eggheads/eggdrop/commit/1ad1eaf0c6047c70f09418b440b947c5e4f12f6f

Discussion: https://github.com/eggheads/eggdrop/pull/1741
--
From 1ad1eaf0c6047c70f09418b440b947c5e4f12f6f Mon Sep 17 00:00:00 2001
From: Michael Ortmann <41313082+michaelortmann@users.noreply.github.com>
Date: Sun, 2 Feb 2025 17:00:41 +0100
Subject: [PATCH] Add -std=gnu99, if available, to CFLAGS

Found by: robert-scheck
Patch by: michaelortmann
Fixes: #1740

GCC 15 changes the default language version for C compilation from -std=gnu17 to -std=gnu23. See https://gcc.gnu.org/gcc-15/changes.html
---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index ed9e1e5b5..0b27c4d06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,6 +54,7 @@ AC_USE_SYSTEM_EXTENSIONS
 EGG_CHECK_ICC
 EGG_CHECK_CCPIPE
 EGG_CHECK_CCWALL
+AX_CHECK_COMPILE_FLAG([-std=gnu99], [CFLAGS="-std=gnu99 $CFLAGS"])
 
 
 # Check for executable extension.
